### PR TITLE
vector homework

### PR DESCRIPTION
**Pull request by meszenab**

### CODE REVIEW

File: course5/homework5/vector.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#include "vector.h"
+#include <iostream>
+#include <cmath>
+
+#define Pi 3.14159265
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: cmath provides the constant

File: course5/homework5/vector.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    Vector2D operator*(double r, const Vector2D& V)
+    {
+        Vector2D W(r * V.GetX(), r * V.GetY());
+        return W;
+    }
+
+}
+
+int main(int, char**)
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: As per assignment description: please, put main into a separate file

File: course5/homework5/vector.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        //    Vector2D(Vector2D&&);
+            Vector2D& operator=(const Vector2D&);
+        //    Vector2D&& operator=(Vector2D&& src);
+            double GetX() const;
+            double GetY() const;
+            void SetX(double);
+            void SetY(double);
+            double GetR() const;
+            double GetAngle() const;
+            Vector2D operator+(const Vector2D&);
```
Comment (maciekgajewski) [inline course5/homework5/vector.h]: Is there a reason why operator+ can't be const?

File: course5/homework5/vector.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            Vector2D operator-(const Vector2D&);
```
Comment (maciekgajewski) [inline course5/homework5/vector.h]: Unary minus operator is missing

File: course5/homework5/vector.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            Vector2D& operator+=(const Vector2D&);
+            Vector2D& operator-=(const Vector2D&);
+            bool operator==(const Vector2D&);
+            bool operator!=(const Vector2D&);
```
Comment (maciekgajewski) [inline course5/homework5/vector.h]: these should be const too

File: course5/homework5/vector.cc [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+#define Pi 3.14159265
+
+namespace vector{
+
+    Vector2D::Vector2D(double x, double y) : x(x), y(y) {}
+    Vector2D::Vector2D(const Vector2D& src) : x(src.x), y(src.y)  {}
+    Vector2D& Vector2D::operator=(const Vector2D& src)
+    { 
+        x = src.x;
+        y = src.y;
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc:13]: something's wrong with this line

File: course5/homework5/vector.cc [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    {
+        this->x = x;
+    }
+
+    void Vector2D::SetY(double y)
+    {
+        this->y = y;
+    }
+
+    double Vector2D::GetR() const { return sqrt(x*x + y*y); }
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: no way this compiles!

File: course5/homework5/vector.cc [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+    Vector2D operator "" _x(long double x)
+    {
+        Vector2D X(x, 0);
+        return X;
+    }
+
+    Vector2D operator "" _y(long double y)
+    {
+        Vector2D Y(0, y);
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc:82]: broken intentation

File: course5/homework5/vector.cc [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        return Y;
+    }
+
+    Vector2D operator "" _phi(long double phi) // user can provide phi in degree
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: Why in degree? All functions in the math lib expect input in radians

File: course5/homework5/vector.cc [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    Vector2D& Vector2D::operator-=(const Vector2D& src)
+    {
+        x -= src.x;
+        y -= src.y;
+        return *this;
+    }
+
+    bool Vector2D::operator==(const Vector2D& src)
+    {
+        if ( x == src.x && y == src.y )
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: the only purpose of the entire 4-line if statement id to convert bool to bool

File: course5/homework5/vector.cc (hunk 7 above)
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: Please, qualify the library functions: ```std::sqrt```, ```std::sin``` etc

File: course5/homework5/vector.cc [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+
+    std::cout << "u = " << u << std::endl;
+
+    v += u;
+
+    std::cout << "v = " << v << std::endl;
+
+    std::cout << "v == u: " << (v == u) << std::endl;
+
+    std::cout << "v != u: " << (v != u) << std::endl;
```
Comment (maciekgajewski) [inline course5/homework5/vector.cc]: Many of the methods remain untested, for instance the getters for radial coords.

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
- put main into a separate file
- add unary minus implementation
- expand tests to cover all methods

File: course5/homework5/vector.cc (hunk 7 above)
Comment (meszenab) [reply course5/homework5/vector.cc]: It does compile and gives the expected result. What make you say that? sqrt vs. std::sqrt?

File: course5/homework5/vector.h (hunk 3 above)
Comment (meszenab) [reply course5/homework5/vector.h]: You mean that the method is const, so
Vector2D operator+(const Vector2D&) const; ?

File: course5/homework5/vector.cc (hunk 6 above)
Comment (meszenab) [reply course5/homework5/vector.cc:13]: Hm, seems to work for me. Could you elaborate?

File: course5/homework5/vector.cc (hunk 7 above)
Comment (maciekgajewski) [reply course5/homework5/vector.cc]: You must have missed this part of the lecture:
```sqrt``` is a C function that compiles and works here sort of by accident. But C++ only guarantees ```std::sqrt```. There may be a C+ compiler now, or in the future, that will fail to compile the code.

File: course5/homework5/vector.cc (hunk 6 above)
Comment (maciekgajewski) [reply course5/homework5/vector.cc:13]: I looked differently when I was looking a it. Maybe github was tricking me...?

File: course5/homework5/vector.h (hunk 3 above)
Comment (maciekgajewski) [reply course5/homework5/vector.h]: yes

**Pull request closed without merging**